### ISSUE TITLE
Constants for defined values

### ISSUE THREAD

**Issue by awharkrider** [zendesk/sunshine-conversations-java#8]
This issue is more of a suggested improvement than a direct issue. 

For objects that have defined values, the SDK would be easier to use if the values were constants such as Java enums.  It may also fit with good programming standards to use constants instead of passing strings around where defined values are expeted. 

I'm sure there could be other defined values that would work well as constants, but the two that I have dealt directly with so far are:  messaging "type" and integration "status".

Messaging "type" would be beneficial to have defined as constants for supported types. In my implementation we have something like this:
```
/**
 * Smooch messaging type enum. Supported messaging platforms
 */
public enum MessageType {
    
    TWITTER("twitter"),
    FACEBOOK("messenger"),
    LINE("line");
    
    private final String type;
    
    MessageType(String type) {
        this.type = type;
    }
    
    public String getType() {
        return type;
    }
}
```

For the integration "status"  I defined them like this based of the documentation possible values: 
```
/**
 *  Possible values for a smooch integration status.
 *
 *  https://github.com/smooch/smooch-java/blob/master/docs/Integration.md
 */
public enum IntegrationStatus {
    
    ACTIVE("active"),
    INACTIVE("inactive"),
    ERROR("error");
    
    private final String status;
    
    IntegrationStatus(String status) {
        this.status = status;
    }
    
    public String getStatus() {
        return status;
    }
}
```

**Comment by mspensieri** [comment]
Thanks for the suggestion. The code in this library is auto-generated, so depending on the tooling it might not be possible to make improvements like this, but we'll look into whether or not it's possible and update this issue

**Comment by maxlacoste92** [comment]
We included a new enums file in [v5.0](https://github.com/smooch/smooch-java/releases/tag/5.0) to be used as reference.
